airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -232,20 +233,18 @@ public AirbyteCatalog discover(final JsonNode config) {
 
   @Override
   public List<TableInfo<CommonField<JDBCType>>> discoverInternal(final JdbcDatabase database) throws Exception {
-    final List<TableInfo<CommonField<JDBCType>>> internals = super.discoverInternal(database);
     if (schemas != null && !schemas.isEmpty()) {
-      // process explicitly filtered (from UI) schemas
-      final List<TableInfo<CommonField<JDBCType>>> resultInternals = internals
-          .stream()
-          .filter(this::isTableInRequestedSchema)
-          .toList();
-      for (final TableInfo<CommonField<JDBCType>> info : resultInternals) {
-        LOGGER.debug("Found table (schema: {}): {}", info.getNameSpace(), info.getName());
-      }
-      return resultInternals;
+      return schemas.stream().flatMap(schema -> {
+        LOGGER.info("Get columns for schema: {}", schema);
+        try {
+          return super.discoverInternal(database, schema).stream();
+        } catch (Exception e) {
+          throw new ConfigErrorException(String.format("Error getting columns for schema: %s", schema), e);
```

Comment:
@postamar I'm testing this on the instance with a large database and even though I see the discovery error in logs the discovery pod finishes with 0 status and then the airbyte server gets `Uncaught exception
java.lang.NullPointerException: Cannot invoke "io.airbyte.api.model.generated.AirbyteCatalog.getStreams()" because "discovered" is null`

Am I doing something wrong in terms how the exception should be handled and re thrown? Perhaps we should raise `OperationExecutionException` or just `RuntimeException` instead?

Discovery pod logs:
```
2024-08-12 10:57:05 ERROR i.a.w.i.VersionedAirbyteStreamFactory(internalLog):304 - ERROR main i.a.c.i.u.ConnectorExceptionHandler(handleException):80 caught exception! io.airbyte.commons.exceptions.ConfigErrorException: Error getting columns for schema: MY_SUPER_BIG_SCHEMA
        at io.airbyte.integrations.source.mssql.MssqlSource.lambda$discoverInternal$1(MssqlSource.java:242) ~[io.airbyte.airbyte-integrations.connectors-source-mssql.jar:?]
...

2024-08-12 10:57:06 INFO i.a.c.ConnectorWatcher(run):136 - Marking workload as successful
2024-08-12 10:57:06 INFO i.a.c.i.LineGobbler(voidCall):166 - 
2024-08-12 10:57:06 INFO i.a.c.i.LineGobbler(voidCall):166 - ----- END DISCOVER -----
2024-08-12 10:57:06 INFO i.a.c.i.LineGobbler(voidCall):166 - 
2024-08-12 10:57:06 INFO i.a.c.ConnectorWatcher(exitProperly):189 - Deliberately exiting process with code 0.
```

and in airbyte-server: 
```
2024-08-12 10:57:08 ERROR i.a.c.s.e.h.UncaughtExceptionHandler(handle):31 - Uncaught exception
java.lang.NullPointerException: Cannot invoke "io.airbyte.api.model.generated.AirbyteCatalog.getStreams()" because "discovered" is null
```

Reply:
Thanks for taking the time to run this additional test @mkrawc. I don't think you're doing anything wrong. We changed the exception handling logic recently in the `ConnectorExceptionHandler` class. From what I can tell it should be exiting with code 1 when handling a `ConfigErrorException`. Evidently, it's not!? Thoughts @theyueli ?

Reply:
@postamar @theyueli Any thoughts? If no I think we can still merge as.

Reply:
@mkrawc @postamar our current handling of `ConfigErrorException` make sure we exit with code `1`

https://github.com/airbytehq/airbyte/blob/db38025d0fc6f8c9fcca610b70f3c28f8b1d0fa6/airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/util/ConnectorExceptionHandler.kt#L98

To verify, could you use debugger to step through and see if the execution falls into that branch?